src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2893,6 +2895,31 @@ export class ResultsViewPageStore {
         },
         []
     );
+    
+    readonly facetsClinicalDataForSamples = remoteData<ClinicalData[]>({
+        await: () => [
+            this.studies,
+            this.samples
+        ],
+        invoke: () => this.getClinicalData("SAMPLE", this.studies.result!, this.samples.result, ["FACETS_WGD", "FACETS_PURITY"])
+    }, []);
+    
+    @computed get sampleIds(): string[] {
```

Comment:
might be better to make this a `readonly` `remoteData`, and await `this.samples`

Reply:
that's weird, we ran into this very problem this week.    but yes, we should not hide the status of the underlying async data in this way.   developers will assume the do not have to await samples.    also, we can probably just do this operation directly in the logic rather than keep it in computed.   it's not that heavy

Reply:
has this one been resolved?

Reply:
I still think that it might be better to make this a `readonly` `remoteData`

Reply:
@onursumer I'm still not very clear when to use a remoteData or computed in this case. clinicalDataGroupedBySampleMap in this pr also could be a computed since it really is just a map based on the clinical data. I don't think you need a status for that.

Reply:
By using `remoteData` we could make sure that we wait until `this.samples` resolves.  When using `@computed` there is no easy way to make sure that the underlying data is actually there.

Reply:
@onursumer I get that, but for these two examples, they are not really need to wait for the clinical data, the clinical data has the default. The computed really works better. For places need the status, they need to wait for the clinical data.

Reply:
@onursumer @zhx828 @alisman We were just following a pattern we saw in two other places:

```
cbioportal-frontend> grep -r "sampleIds()" src/
src//pages/patientView/clinicalInformation/PatientViewPageStore.ts:    @computed get sampleIds(): string[] {
src//pages/staticPages/tools/oncoprinter/OncoprinterStore.ts:    @computed get sampleIds() {
src//pages/resultsView/ResultsViewPageStore.ts:    @computed get sampleIds(): string[] {
```

Perhaps we can leave it as it is and all three could be refactored in the same way by someone who is more of an expert?